File: Go/1_TwoSum.go
package main

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	var missingIndex int
	var missingExist bool

	var index int
	var value int
	for index, value = range nums {
		missingIndex, missingExist = m[target-value]
		if missingExist {
			awnsear := []int{index, missingIndex}
			return awnsear
		}
		m[value] = index
	}
	awnsear := []int{index, missingIndex}
	return awnsear
}

File: Python/1_TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict_vistos = {}

        for indice, num in enumerate(nums):
            restante = target - nums[indice]
            if restante in dict_vistos:
                return [indice, dict_vistos[restante]]
            else:
                dict_vistos[num] = indice